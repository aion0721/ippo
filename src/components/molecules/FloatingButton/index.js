import React from 'react';
import propTypes from 'prop-types';
import styled from 'styled-components';
import RoundButton from 'src/components/atoms/RoundButton';

const StyledRoundButton = styled(RoundButton)`
  position: fixed;
  right: 16px;
  bottom: 16px;
`;

const FloatingButton = ({ icon, onClick }) => (
  <StyledRoundButton icon={icon} onClick={onClick} large />
);

FloatingButton.displayName = 'FloatingButton';

FloatingButton.propTypes = {
  icon: propTypes.string.isRequired,
  onClick: propTypes.func.isRequired,
};

export default FloatingButton;
