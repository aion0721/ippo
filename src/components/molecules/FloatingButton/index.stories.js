import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import FloatingButton from '.';

const stories = storiesOf('molecules/FloatingButton', module);

stories.add('通常パターン', () => (
  <FloatingButton icon={text('icon', 'notifications')} onClick={action('onClick')} />
));
