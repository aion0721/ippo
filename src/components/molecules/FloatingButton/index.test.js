import React from 'react';
import { snapshot } from 'test/helpers';
import FloatingButton from '.';

const props = {
  icon: 'notifications',
  onClick: jest.fn(),
};

snapshot('FloatingButton/nomal', <FloatingButton {...props} />);
