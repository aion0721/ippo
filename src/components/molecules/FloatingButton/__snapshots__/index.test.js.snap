// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: FloatingButton/nomal 1`] = `
.c1 {
  border-radius: 50%;
  height: 50px;
  width: 50px;
}

.c0 {
  position: fixed;
  right: 16px;
  bottom: 16px;
}

<FloatingButton
  icon="notifications"
  onClick={[MockFunction]}
>
  <Styled(RoundButton)
    icon="notifications"
    onClick={[MockFunction]}
    xlarge={true}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  position: fixed;
  right: 16px;
  bottom: 16px;
",
            ],
          },
          "displayName": "Styled(RoundButton)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      forwardedRef={null}
      icon="notifications"
      onClick={[MockFunction]}
      xlarge={true}
    >
      <RoundButton
        className="c0"
        icon="notifications"
        onClick={[MockFunction]}
        xlarge={true}
      >
        <Styled(Blueprint3.Button)
          className="c0"
          icon="notifications"
          onClick={[MockFunction]}
          xlarge={1}
        >
          <StyledComponent
            className="c0"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  border-radius: 50%;
  height: ",
                    [Function],
                    ";
  width: ",
                    [Function],
                    ";
",
                  ],
                },
                "displayName": "Styled(Blueprint3.Button)",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            forwardedRef={null}
            icon="notifications"
            onClick={[MockFunction]}
            xlarge={1}
          >
            <Blueprint3.Button
              className="c0 c1"
              icon="notifications"
              onClick={[MockFunction]}
              xlarge={1}
            >
              <button
                className="bp3-button c0 c1"
                onClick={[MockFunction]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                type="button"
                xlarge={1}
              >
                <Blueprint3.Icon
                  icon="notifications"
                  key="leftIcon"
                >
                  <span
                    className="bp3-icon bp3-icon-notifications"
                    icon="notifications"
                  >
                    <svg
                      data-icon="notifications"
                      height={16}
                      viewBox="0 0 16 16"
                      width={16}
                    >
                      <desc>
                        notifications
                      </desc>
                      <path
                        d="M8 16c1.1 0 2-.9 2-2H6c0 1.1.9 2 2 2zm6-5c-.55 0-1-.45-1-1V6c0-2.43-1.73-4.45-4.02-4.9 0-.04.02-.06.02-.1 0-.55-.45-1-1-1S7 .45 7 1c0 .04.02.06.02.1A4.992 4.992 0 0 0 3 6v4c0 .55-.45 1-1 1s-1 .45-1 1 .45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1z"
                        fillRule="evenodd"
                        key="0"
                      />
                    </svg>
                  </span>
                </Blueprint3.Icon>
                <Blueprint3.Icon
                  key="rightIcon"
                />
              </button>
            </Blueprint3.Button>
          </StyledComponent>
        </Styled(Blueprint3.Button)>
      </RoundButton>
    </StyledComponent>
  </Styled(RoundButton)>
</FloatingButton>
`;
